HOWTO-stage-a-release.md

```diff
@@ -0,0 +1,167 @@
+# HOW TO STAGE A RELEASE
+
+This file documents how to make an OpenSSL release.  Please fix any errors
+you find while doing, or just after, your next release!
+
+Anyone with access to the necessary resources may stage a release.  Reviews
+for doing so isn't necessary, that's done "naturally" as part of publishing,
+see [HOWTO-publish-a-release.md](HOWTO-publish-a-release.md).
+
+# Automation
+
+**Staging releases is becoming automated**, so this document will soon only
+be interesting to know how to perform this manually, should the need arise
+(automation failure, or to stage releases that automation isn't prepared
+for).
+
+This automation is currently still undergoing tests, and isn't quite
+reflected in [HOWTO-publish-a-release.md](HOWTO-publish-a-release.md).
+Updates pending!
+
+# Table of contents
+
+-   [Prerequisites](#prerequisites)
+    -   [Software](#software)
+    -   [Repositories](#repositories)
+    -   [PGP / GnuPG key](#pgp-gnupg-key)
+    -   [SFTP access](#check-your-access)
+    -   [Prepare your repository checkouts](#prepare-your-repository-checkouts)
+-   [Staging tasks](#staging-tasks)
+
+    -   [Generate the tarball and announcement text](#generating-the-tarball-and-announcement-text)
+    -   [Remember the results](#remember-the-results)
+
+# Prerequisites
+
+## Software
+
+Apart from the basic operating system utilities, you must have the following
+programs in you `$PATH`:
+
+- openssl
+- gpg
+- git
+- ssh
+- sftp
+
+(note: this may not be a complete list)
+
+## Repositories
+
+You must have access to the following repositories:
+
+-   `git@github.openssl.org:otc/tools.git`
+
+    This contains the release staging tool.
+
+-   Any of:
+
+    -   `git@github.openssl.org:openssl/openssl.git`
+
+        This is the public source repository, so is only necessary to stage
+        a public release, which are those that haven't reached End-Of-Life
+        yet.
+
+    -   `git@github.openssl.org:openssl/security.git`
+
+        This is the security source repository, where security fixes are
+        staged before being publically released.  It is used as source
+        repository instead of `openssl/openssl` to stage a security
+        release.
+
+    -   `git@github.openssl.org:openssl/premium.git`
+
+        This is the source repository for premium customers, used for both
+        security and non-security releases.
+
+## PGP / GnuPG key
+
+You must have OpenSSL's team key:
+
+    $ gpg --list-secret-key EFC0A467D613CB83C7ED6D30D894E2CE8B3D79F5
+    sec   rsa4096 2014-10-04 [SC] [expires: 2027-03-09]
+          EFC0A467D613CB83C7ED6D30D894E2CE8B3D79F5
+    uid           [ultimate] OpenSSL Security <openssl-security@openssl.org>
+    uid           [ultimate] OpenSSL security team <openssl-security@openssl.org>
+    uid           [ultimate] OpenSSL OMC <openssl-omc@openssl.org>
+    ssb   rsa4096 2014-10-04 [E] [expires: 2024-01-30]
```

Comment:
This key needs to be added to fingerprints.txt and the user IDs properly uploaded to keys.openpgp.org key server.

It cannot be used with the next release yet because it needs to be in fingerprints.txt in an existing release first.

Reply:
> ... and the user IDs properly uploaded to keys.openpgp.org key server.

That part is actually only true if the key is to be searchable by user ID on keys.openpgp.org.  Of course, I'm not against that, but the actual truth is that the key is perfectly usable without them.

People who have signed the key should probably make sure to verify their addresses, though, as the signatures are otherwise very anonymous...

Reply:
For the rest, yeah you are correct...  unfortunately, 'cause I distinctly remember that we talked about use the OpenSSL team key going forward, during more than one meeting.  But maybe I'm ahead of time...

Reply:
This is also very sad, 'cause it means that we will have to sign the tarballs and announcement texts when *publishing* them.  That can of course be done, but we've basically moved churn that we thought we could automate.  Ah well...

Reply:
The more I think about this, the more I realize that it becomes difficult to sign the tarballs when staging them, when staging and publishing become separate tasks (which they do as soon as we start using automated staging and only do the publishing part manually).

The simplest is probably to let this PR wait until we have the team key firmly in place in `doc/fingerprints.txt` in all current release branches.  In other words, let's rush that one in!